packages/trader/src/App/Components/Elements/TogglePositions/toggle-positions-mobile.jsx

```diff
@@ -24,34 +24,36 @@ const TogglePositionsMobile = observer(
         onClickCancel,
         toggleUnsupportedContractModal,
     }) => {
-        const { portfolio, ui } = useStore();
         const { symbol, contract_type: trade_contract_type } = useTraderStore();
-        const { removePositionById: onClickRemove } = portfolio;
-        const { togglePositionsDrawer, is_positions_drawer_on } = ui;
-        let filtered_positions = [];
+        const { togglePositionsDrawer, is_positions_drawer_on } = useStore().ui;
+        const [hidden_positions_ids, setHiddenPositionsIds] = React.useState([]);
+        const filtered_positions = all_positions
+            .filter(
+                p =>
+                    p.contract_info &&
+                    symbol === p.contract_info.underlying &&
+                    filterByContractType(p.contract_info, trade_contract_type) &&
+                    hidden_positions_ids.every(hidden_position_id => hidden_position_id !== p.contract_info.contract_id)
+            )
+            .slice(0, 5);
 
         const closeModal = () => {
-            filtered_positions.slice(0, 5).map(position => {
-                const { contract_info } = position;
-                if (contract_info?.is_sold) {
-                    onClickRemove(contract_info.contract_id);
-                }
-            });
+            setHiddenPositionsIds([
+                ...new Set([
+                    ...hidden_positions_ids,
+                    ...filtered_positions
```

Comment:
Should we rename `filtered_positions` variable, because you're filtering it one more time and does not seem to be a descriptive name.

Reply:
@matin-deriv `filtered_positions` are filtered positions to be displayed in the modal, and at this line i'm just selecting only sold ones from filtered_positions and setting their ids as hidden positions ids using spread operator. Sold positions are the positions we need to hide after we close the modal). :)

Reply:
@matin-deriv or would you like me to create a variable like `sold_positions`?